app/controllers/classifications_controller.rb

```diff
@@ -150,4 +133,22 @@ def message
                raise 'Unsupported state'
              end
   end
+
+  private
+
+  def find_info_request
+    @info_request = InfoRequest.not_embargoed.find_by!(
+      url_title: params[:url_title]
+    )
+  end
+
+  def authorise_info_request
+    # If this is an external request, go to the request page - we don't allow
+    # state change from the front end interface.
+    if @info_request.is_external?
+      redirect_to_info_request
```

Comment:
Looks like this introduces a spec failure:

```
Failures:

  1) ClassificationsController POST create if the request is external should redirect to the request page
     Failure/Error: redirect_to_info_request

     NameError:
       undefined local variable or method `redirect_to_info_request' for #<ClassificationsController:0x00005623971a5548>
     # ./app/controllers/classifications_controller.rb:149:in `authorise_info_request'
     # /home/vagrant/bundle/ruby/2.5.0/gems/activesupport-5.1.7/lib/active_support/callbacks.rb:413:in `block in make_lambda'
     # /home/vagrant/bundle/ruby/2.5.0/gems/activesupport-5.1.7/lib/active_support/callbacks.rb:197:in `block (2 levels) in halting'
     # /home/vagrant/bundle/ruby/2.5.0/gems/actionpack-5.1.7/lib/abstract_controller/callbacks.rb:12:in `block (2 levels) in <module:Callbacks>'
     # /home/vagrant/bundle/ruby/2.5.0/gems/activesupport-5.1.7/lib/active_support/callbacks.rb:198:in `block in halting'
     # /home/vagrant/bundle/ruby/2.5.0/gems/activesupport-5.1.7/lib/active_support/callbacks.rb:507:in `block in invoke_before'
     # /home/vagrant/bundle/ruby/2.5.0/gems/activesupport-5.1.7/lib/active_support/callbacks.rb:507:in `each'
     # /home/vagrant/bundle/ruby/2.5.0/gems/activesupport-5.1.7/lib/active_support/callbacks.rb:507:in `invoke_before'
     # /home/vagrant/bundle/ruby/2.5.0/gems/activesupport-5.1.7/lib/active_support/callbacks.rb:106:in `block in run_callbacks'
     # ./app/controllers/application_controller.rb:116:in `record_memory'
     # /home/vagrant/bundle/ruby/2.5.0/gems/activesupport-5.1.7/lib/active_support/callbacks.rb:117:in `block in run_callbacks'
     # /home/vagrant/bundle/ruby/2.5.0/gems/activesupport-5.1.7/lib/active_support/callbacks.rb:135:in `run_callbacks'
     # /home/vagrant/bundle/ruby/2.5.0/gems/actionpack-5.1.7/lib/abstract_controller/callbacks.rb:19:in `process_action'
     # /home/vagrant/bundle/ruby/2.5.0/gems/actionpack-5.1.7/lib/action_controller/metal/rescue.rb:20:in `process_action'
     # /home/vagrant/bundle/ruby/2.5.0/gems/actionpack-5.1.7/lib/action_controller/metal/instrumentation.rb:32:in `block in process_action'
     # /home/vagrant/bundle/ruby/2.5.0/gems/activesupport-5.1.7/lib/active_support/notifications.rb:166:in `block in instrument'
     # /home/vagrant/bundle/ruby/2.5.0/gems/activesupport-5.1.7/lib/active_support/notifications/instrumenter.rb:21:in `instrument'
     # /home/vagrant/bundle/ruby/2.5.0/gems/activesupport-5.1.7/lib/active_support/notifications.rb:166:in `instrument'
     # /home/vagrant/bundle/ruby/2.5.0/gems/actionpack-5.1.7/lib/action_controller/metal/instrumentation.rb:30:in `process_action'
     # /home/vagrant/bundle/ruby/2.5.0/gems/actionpack-5.1.7/lib/action_controller/metal/params_wrapper.rb:252:in `process_action'
     # /home/vagrant/bundle/ruby/2.5.0/gems/activerecord-5.1.7/lib/active_record/railties/controller_runtime.rb:22:in `process_action'
     # /home/vagrant/bundle/ruby/2.5.0/gems/actionpack-5.1.7/lib/abstract_controller/base.rb:124:in `process'
     # /home/vagrant/bundle/ruby/2.5.0/gems/actionview-5.1.7/lib/action_view/rendering.rb:30:in `process'
     # /home/vagrant/bundle/ruby/2.5.0/gems/actionpack-5.1.7/lib/action_controller/metal.rb:189:in `dispatch'
     # /home/vagrant/bundle/ruby/2.5.0/gems/actionpack-5.1.7/lib/action_controller/test_case.rb:513:in `process'
     # /home/vagrant/bundle/ruby/2.5.0/gems/rails-controller-testing-1.0.2/lib/rails/controller/testing/template_assertions.rb:61:in `process'
     # /home/vagrant/bundle/ruby/2.5.0/gems/actionpack-5.1.7/lib/action_controller/test_case.rb:400:in `post'
     # /home/vagrant/bundle/ruby/2.5.0/gems/rails-controller-testing-1.0.2/lib/rails/controller/testing/integration.rb:12:in `block (2 levels) in <module:Integration>'
     # ./spec/controllers/classifications_controller_spec.rb:9:in `block (4 levels) in <top (required)>'

Finished in 2 minutes 4.3 seconds (files took 28.47 seconds to load)
79 examples, 1 failure

Failed examples:

rspec ./spec/controllers/classifications_controller_spec.rb:8 # ClassificationsController POST create if the request is external should redirect to the request page
```

Reply:
Ah yes, that was added in the wrong commit - I've pushed two fixups to remove here and add to the right commit.